classes/classes/Scenes/Areas/GlacialRift/AmarokScene.as

```diff
@@ -0,0 +1,77 @@
+			amarokFlagCheck();
+			}
+		}
+
+		public function amarokFlagCheck():void {
+			if (player.findStatusEffect(StatusEffects.Infested) >= 0) {
+					amarokNoEat();
+			}
+			else if (flags[kFLAGS.AMAROK_LOSSES] >= 5) {
+					amarokBadEnd();
+			}
+			else if (flags[kFLAGS.AMAROK_LOSSES] < 5) {
+					amarokChompo();
+			}
+		}
+		
+		private function amarokNoEat():void { //He doesn't want to eat worms
+			clearOutput();
+			outputText("\n\nMuch to your surprise, however, the Amarok suddenly backs away with a low growl. It resumes sniffing you at a distance, eventually coming over to your lower half. The worms inside you wriggle about, and you let out a low groan of discomfort. As though on cue, the Amarok take a sniff of your crotch, then promptly backs off with another snarl. It must be able to detect your worm infestation, and more importantly, not like it! It turns away and kicks snow on you with one of its back back, letting out a huff of irritation. You watch it walk off as blackness washes over your vision.", false);
+			combat.cleanupAfterCombat();
+		}
+		
+		private function amarokChompo():void {
+			combat.cleanupAfterCombat();
+			clearOutput();
+			outputText("It lets out a soft bark as you close your eyes and skips to the other side of you. You remain still in fear of what it plans to do. It picks you by your neck. Its grip isn't hard enough to even break your skin, but firm enough that it won't drop you. It then begins to drag you somewhere. Something in your mind tells you to struggle, but you can't seem to muster the will to try. It'd just bite through your neck and kill you anyway. The feeling of its hot breath running down your back is oddly soothing. You open your eyes long enough to make sense of your surroundings, then pass out.", false);
+			outputText("\n\nYou awaken some time later to a chorus of yips. You've stopped moving. You're on the ground, with small paws and dull teeth prodding you all over. They don't hurt you outside of being uncomfortable, but you're sure you'll be left with bruises. You focus your vision and slowly make out where you are-- the side of a short cliff, protected from snow by tall trees. You're laying atop of pine needles and dead leaves. A strong scent of feces causes you to jolt upright in disgust, and something that'd been gnawing on your ear fell to the ground with a squeak. You look back at it. It's a black wolf pup, likely the Amarok's. You spot three others scattered about the den. You must have been left as food for them. The pups, however, seem much more interested in using you as a toy than a meal. You get to your feet, shaking off the pups. You don't feel like sticking around for them to change their minds.", false);
+			outputText("\n\nYou spot your " + player.armorName + " a bit away. It must've been ripped off you when you were passed out. You scramble to it and put it back on, then head back to camp as fast as you can.", false);
+			doNext(camp.returnToCampUseOneHour);
```

Comment:
Do not use doNext and combat.cleanupAfterCombat this way. Use combat.cleanupAfterCombat at the end of the function.